src/codegen/llvm/codegen_llvm_helper_visitor.cpp

```diff
@@ -611,17 +642,30 @@ void CodegenLLVMHelperVisitor::visit_nrn_state_block(ast::NrnStateBlock& node) {
     }
 
     /// remainder loop possibly vectorized on vector_width
-    {
+    if (vector_width > 1) {
         /// loop constructs : initialization, condition and increment
-        const auto& condition = create_expression("{} < {}"_format(INDUCTION_VAR, NODECOUNT_VAR));
-        const auto& increment = loop_increment_expression(INDUCTION_VAR, 1);
+        const auto& condition =
+            loop_count_expression(INDUCTION_VAR, NODECOUNT_VAR, /*vector_width=*/1);
+        const auto& increment = loop_increment_expression(INDUCTION_VAR, /*vector_width=*/1);
 
         /// convert local statement to codegenvar statement
         convert_local_statement(*loop_block);
 
         auto for_loop_statement_remainder =
             std::make_shared<ast::CodegenForStatement>(nullptr, condition, increment, loop_block);
 
+        const auto& loop_statements = for_loop_statement_remainder->get_statement_block();
+        // \todo: Change RenameVisitor to take a vector of names to which it would append a single
+        // prefix.
+        for (const auto& name: int_variables) {
+            visitor::RenameVisitor v(name, epilogue_variable_prefix + name);
+            loop_statements->accept(v);
+        }
+        for (const auto& name: double_variables) {
+            visitor::RenameVisitor v(name, epilogue_variable_prefix + name);
+            loop_statements->accept(v);
+        }
```

Comment:
@georgemitenkov : could you remind myself why we need to rename variables? I think mentioned it but I vaguely remember - variables need to have unique names across different basic blocks?